# HG changeset patch
# User Carlo de Falco <kingcrimson@tiscali.it>
# Date 1315065706 -7200
#      Sat Sep 03 18:01:46 2011 +0200
# Node ID 69a4609e61e2700ebefee8258473aef308324e9e
# Parent  b2aca5ac54a2ca021e03e1e7fc114bfa73330414
codesprint: Add a test to orth.m

* orth.m : Add a test.

diff --git a/scripts/linear-algebra/orth.m b/scripts/linear-algebra/orth.m
--- a/scripts/linear-algebra/orth.m
+++ b/scripts/linear-algebra/orth.m
@@ -74,8 +74,17 @@ function retval = orth (A, tol)
 
   else
 
     print_usage ();
 
   endif
 
 endfunction
+
+%!test
+%! for ii=1:20
+%!   A = rand (10, 10);
+%!   V = orth (A);
+%!   if (det (A) != 0)
+%!     assert (V'*V, eye (10), 100*eps)
+%!   endif
+%! endfor
